### PR TITLE
fix: use two in memory indexes, for resolve and builds

### PR DESCRIPTION
**Pull request by tdejager**
This PR makes the resolver use two memory indexes, one for the resolution process and one for the potential building of wheels. Even though uv uses different internal revolvers it was sometimes still getting data that was overridden by us during wheel builds, which was unexpected.

This was cropping up in issues like #1686. We've made a minimal version of the lock file where this problem was occuring. And added that as a test.

Fixes https://github.com/prefix-dev/pixi/pull/1686

### CODE REVIEW

**Review by baszalmstra — APPROVED** [review]

**Comment by tdejager** [comment]
Seems to be some other problem cropping up, converting into draft.

**Comment by tdejager** [comment]
Seemed to be a corrupted cargo git cache, never had that one before.. Back to undrafting.

File: tests/install_tests.rs [hunk 1]
```diff
@@ -661,6 +661,31 @@ setup(
...
+        r#"
+        [project]
+        channels = ["conda-forge"]
+        name = "pixi-source-problem"
+        platforms = ["{platform}"]
+
+        [dependencies]
+        pip = ">=24.0,<25"
+
+        # The transitive dependency of viser was causing issues
```
Comment (Hofer-Julian) [inline tests/install_tests.rs:679]:
```suggestion
        # The transitive dependencies of viser was causing issues
```

**Review by Hofer-Julian — APPROVED** [review]
One nitpick

**Comment by baszalmstra** [comment]
I fixed these failing tests in #1973 as well.

**Pull request merged**